### ISSUE TITLE
Disabled buttons with .active are incorrectly styled by :hover styles

### ISSUE THREAD

**Issue by toby-howell** [twbs/bootstrap#21235]
The following CSS is what is applied when a `button.btn.btn-secondary.active[disabled]` is hovered over:

```css
.btn-secondary.disabled:hover, .btn-secondary:disabled:hover {
    background-color: #FFFFFF;
    border-color: #CCCCCC;
}
```

This overrides declarations applied by `.active`:

```css
.btn-secondary.active, .btn-secondary:active, .open > .btn-secondary.dropdown-toggle {
    background-color: #E6E6E6;
    border-color: #ADADAD;
    /* ... */
}
```

This makes the button appear to be responsive to the user's interaction, despite it being disabled.

An example of the change needed is as follows:

```css
.btn-secondary.active.disabled:hover, .btn-secondary.active:disabled:hover {
    background-color: #E6E6E6;
    border-color: #ADADAD;
}
```

- **Operating system:** Windows 10 Pro x64
- **Browser:** Chrome 54.0.2840.99 m
- **Test case and workaround:** [https://jsbin.com/fakejeyero](https://jsbin.com/fakejeyero/edit?css,output)

Although I used `.btn-secondary` as an example, this affects all similarly coloured button styles, as well as other button styles, such as outline buttons (although I have not exhaustively tested other styles). This issue was encountered in Bootstrap v4.0.0-alpha.5.

**Comment by cvrebert** [comment]
I kinda question why we have `:disabled:hover` styles in the first place. If a button is disabled, then it's not interactive. So why are we making it interact-able via hovering?
It doesn't seem like native buttons have such an interaction.

**Comment by patrickhlauke** [comment]
agree that `:disabled:hover` is confusing/unnecessary

**Comment by mdo** [comment]
Okay, digging into this again I think the best path forward is a simpler one—no hover interactions on disabled *or* active elements. Native controls work this way overall I think, but it's still worth discussing. Opening a PR shortly.

**Comment by mdo** [comment]
Merged #21439 to address this.
